### PR TITLE
 Extract common code from request function

### PR DESCRIPTION
**Pull request by LukasPietzschmann**

### CODE REVIEW

**Comment by LukasPietzschmann** [comment]
I think I could also implement a `create_request_function` function, which can return (or _generate_) a function like `request_slice`. But I'm struggling with R at the moment, so this is not yet ready xD

**Comment by EagleoutIce** [comment]
i think there is no need to overengineer this right now

**Comment by EagleoutIce** [comment]
:+1: so simply (re-)ping me when you're done with whatever decision you come to :)

**Comment by LukasPietzschmann** [comment]
@EagleoutIce 🛎️ (re)ping :)

I didn't manage to get it to work. So feel free to merge what I got so far.

File: R/utils.R [hunk 1]
```diff
@@ -30,3 +30,32 @@ get_filetoken <- function(filepath = rlang::missing_arg(), content = rlang::miss
...
+
+#' Checks if the given response was successful or not.
+#'
+#' The function always returns a list with the response's ID. In case of an
+#' error, the list also contains the errors reason. In case of a successful
+#' response, the list also contains the complete response.
+#'
+#' @param res The response to handle
+#'
+#' @return A list with either the error's reason or the complete response
```
Comment (EagleoutIce) [inline R/utils.R:42]: Maybe add the name of the list elements? as we have no structures to back them for the moment?

**Review by EagleoutIce — APPROVED** [review]

**Pull request merged**